### PR TITLE
Add min/max limits for enumerator question answers

### LINKED ISSUE #7136: Allow admin to specify a minimum/maximum answers for enumerator questions
Example: Applicant can choose to have up to 3 trees. Each tree option has multiple fields that would be part of the enumerator.

Currently enumerators are unbounded.

We should add a setting to allow specifying the maximum number of times an enumerator can be filled out. Similarly a minimum setting may be useful.

### PR DESCRIPTION
**Pull request by jeffpw-goog**
### Description

Add a setting to enumerator questions that allows the admin to specify a minimum and maximum limit for the number of answers.

#### General

Read the full guidelines for PRs [here](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#creating-a-pull-request)

- [x] Added the correct label: < feature | enhancement | bug | under-development | dependencies | infrastructure | ignore-for-release | database >
- [x] Assigned to a specific person, `civiform/developers`, or a [more specific round-robin list](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#adding-reviewers)
- [x] Added an additional reviewer from outside your organization as FYI (if the primary reviewer is in the same organization as you)
- [x] Created unit and/or browser tests which fail without the change (if possible)
- [x] Performed manual testing (Chrome and Firefox if it includes front-end changes)
- [x] Extended the README / documentation, if necessary - https://github.com/civiform/docs/pull/471

#### User visible changes

- [x] Followed steps to [internationalize new strings](https://github.com/civiform/civiform/wiki/Internationalization-%28i18n%29#internationalization-for-application-strings)
  - [x] Added context strings to new [messages](https://github.com/civiform/civiform/blob/main/server/conf/i18n/messages)
  - [x] Didn't use a message in applicant facing code that isn't translated yet (unless behind a flag)
- [x] Wrote browser tests using the [validateAccessibility](https://sourcegraph.com/github.com/civiform/civiform/-/blob/browser-test/src/support/index.ts?L437:14&subtree=true) method
- [x] Tested on mobile view. See [mobile device mode](https://developer.chrome.com/docs/devtools/device-mode/)
- [x] Manually tested at 200% size
- [x] Manually evaluated tab order

### Issue(s) this completes

Fixes #7136

### CODE REVIEW

**Review by gwendolyngoetz — COMMENTED** [review]
Maybe I missed it. What happens if I set the min to be larger than than max on the admin question page?

File: server/test/services/applicant/question/EnumeratorQuestionTest.java [hunk 1]
```diff
@@ -122,6 +136,50 @@ public void withDuplicateNames_hasValidationErrors() {
                         MessageKey.ENUMERATOR_VALIDATION_DUPLICATE_ENTITY_NAME))));
   }
```
Comment (dkatzz) [inline server/test/services/applicant/question/EnumeratorQuestionTest.java:138]: **suggestion:** Could we add one more test that it passes validation when they set a min / max and the number of items a person adds is within that?

**Comment by jeffpw-goog** [comment]
> Maybe I missed it. What happens if I set the min to be larger than than max on the admin question page?

Good point! There's currently no handling for this.  Looking at all the other question types that have min/max, only the checkbox question has validation, here: https://github.com/civiform/civiform/blob/91bdeac8aa7e4ed226b3615ac0fbc7e7a0a9455e/server/app/services/question/types/QuestionDefinition.java#L279-L378

I didn't see it because it's in the base class rather than the concrete question class (and I would still have had to be imitating MultiOptionQuestionDefinition rather than one of the others).

Seems like the proper way to do this would be to add a validate() method for concrete question definitions to override if they want to, and then to add validation for all questions that have min/max. Really we should be using composition instead of inheritance, but I don't think I want to go down that rabbit hole.

Do you want to block this PR on adding validation? In that case I'll do a separate PR first to create the overridable validate method, move the question-specific validation logic from QuestionDefinition to the subclasses (EnumeratorQuestion has a bit too), and add validation for the other existing question types that should have it. Otherwise I can file a new issue to handle it for all questions as tech debt.

**Comment by jeffpw-goog** [comment]
> If someone has already answered an enum question with 5 answers and then the question gets updated to say max is 4, what happens? Do they get a validation error when they next open the question?

Yes, they get a validation error when they open the question and then try to save.

File: server/test/services/applicant/question/EnumeratorQuestionTest.java (hunk 1 above)
Comment (jeffpw-goog) [reply server/test/services/applicant/question/EnumeratorQuestionTest.java:138]: Good idea - added!

**Comment by gwendolyngoetz** [comment]
If nothing else on the admin side is currently checking min/max we can file a bug and do them in a separate PR

**Comment by dkatzz** [comment]
> > If nothing else on the admin side is currently checking min/max we can file a bug and do them in a separate PR
> 
> Filed #7919 to follow up on this.

Thank you!! But yeah agreed - I think this would be good to do, but can be done later as a follow up

**Pull request merged**